### PR TITLE
fix: [CDS-34422]: Fixed the yaml view for parallel stage in triggers

### PR DESCRIPTION
**Pull request by goyalsonali**
Summary:

 Jira Links:

https://harness.atlassian.net/browse/CDS-34422

Screenshots:
earlier:

Screenshot 2022-05-04 at 12 21 02 PM

now:
Screenshot 2022-05-12 at 3 27 12 PM

https://user-images.githubusercontent.com/98508399/166635825-0d66645e-6533-427d-b42b-d0600f4a308b.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by goyalsonali** [comment]
retrigger titlecheck

**Comment by goyalsonali** [comment]
retrigger titlecheck

**Comment by goyalsonali** [comment]
retrigger titlecheck

**Comment by goyalsonali** [comment]
trigger rebase

**Comment by goyalsonali** [comment]
trigger rebase

**Review by devinder-harness — APPROVED** [review]

**Comment by goyalsonali** [comment]
retrigger jest

**Comment by goyalsonali** [comment]
retrigger jest

**Comment by mtran7** [comment]
@goyalsonali In the video, the connectorRef has become empty. We need to pull this value too into the stage. 

Also can you test using propagate from existing stage and see if it works as expected? I'd reach out to a PM if you have questions too as I'm not 100% sure what would happen

**Review by shamala-harness — DISMISSED** [review]

**Comment by goyalsonali** [comment]
retrigger jest

**Comment by goyalsonali** [comment]
retrigger jest

File: src/modules/72-triggers/pages/triggers/TriggersWizardPage.tsx [hunk 1]
```diff
@@ -458,12 +461,33 @@ const TriggersWizardPage: React.FC = (): JSX.Element => {
...
     if (onEditInitialValues?.pipeline && template?.data?.inputSetTemplateYaml && mergedPipelineKey < 1) {
-      const newOnEditPipeline = merge(
+      let newOnEditPipeline = merge(
         parse(template?.data?.inputSetTemplateYaml || '')?.pipeline,
         onEditInitialValues.pipeline || {}
       )
+
+      /*this check is needed as when trigger is already present with 1 stage and then tries to add parallel stage,
+      we need to have correct yaml with both stages as a part of parallel*/
+      if (
```
[the remark this answers is not in the document]
Comment (mtran7) [reply src/modules/72-triggers/pages/triggers/TriggersWizardPage.tsx:484]: And see if we can add the same warning message if the user goes from parallel stages -> single stage

[the remark this answers is not in the document]
Comment (goyalsonali) [reply src/modules/72-triggers/pages/triggers/TriggersWizardPage.tsx:484]: Yes, it is working fine and warning message is shown in both cases.

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**